client/my-sites/checkout/composite-checkout/lib/pending-page.ts

```diff
@@ -0,0 +1,68 @@
+import page from 'page';
+
+/**
+ * Redirect to a checkout pending page and from there to a (relative or absolute) url.
+ */
+export function redirectThroughPending(
+	url: string,
+	siteSlug: string | undefined,
+	orderId: string | number | undefined
+): void {
+	if ( ! isRelativeUrl( url ) ) {
+		return absoluteRedirectThroughPending( url, siteSlug, orderId );
+	}
+	try {
+		relativeRedirectThroughPending( url, siteSlug, orderId );
+	} catch ( err ) {
+		absoluteRedirectThroughPending( url, siteSlug, orderId );
+	}
+}
+
+function isRelativeUrl( url: string ): boolean {
+	return url.startsWith( '/' ) && ! url.startsWith( '//' );
+}
+
+/**
+ * Redirect to a checkout pending page and from there to a relative url.
+ */
+export function relativeRedirectThroughPending(
+	url: string,
+	siteSlug: string | undefined,
+	orderId: string | number | undefined
+): void {
+	window.scrollTo( 0, 0 );
+	page( addUrlToPendingPageRedirect( url, siteSlug, orderId, 'relative' ) );
+}
+
+/**
+ * Redirect to a checkout pending page and from there to an absolute url.
+ */
+export function absoluteRedirectThroughPending(
+	url: string,
+	siteSlug: string | undefined,
+	orderId: string | number | undefined
+): void {
+	window.location.href = addUrlToPendingPageRedirect( url, siteSlug, orderId, 'absolute' );
+}
+
+/**
+ * Add a relative or absolute url to the checkout pending page url.
+ */
+export function addUrlToPendingPageRedirect(
```

Comment:
It'd be nice to have unit tests for this particular function, although its dependency on `window` may make it complicated.

Reply:
Good idea.